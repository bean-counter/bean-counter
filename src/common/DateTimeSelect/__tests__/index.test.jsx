import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import DateTimeSelect from '../'

describe('DateTimeSelect', () => {
  const mochOnChange = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot', () => {
    const component = renderer.create((
      <DateTimeSelect
        label="Date"
        name="createdAt"
        value={new Date('September 20th 11:36 a.m.')}
        onChange={mochOnChange}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  // it('handles checnges the date', () => {
  //   const wrapper = mount((
  //     <DateTimeSelect
  //       label="Date"
  //       name="createdAt"
  //       value={new Date('September 20th 11:36 a.m.')}
  //       onChange={mochOnChange}
  //     />
  //   ))
  //   console.log(wrapper)
  //   // expect(wrapper.find('button > span').text()).toBe('Login')
  //   // wrapper.find('button').simulate('click')
  //   // expect(mochHandleLogin).toHaveBeenCalled()
  //   // expect(mochHandleLogout).not.toHaveBeenCalled()
  // })
})
