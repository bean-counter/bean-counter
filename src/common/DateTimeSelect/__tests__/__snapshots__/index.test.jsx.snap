// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimeSelect matches snapshot 1`] = `
<div
  aria-describedby={undefined}
  className="MuiFormControl-root-2"
  onClick={[Function]}
  onError={undefined}
  onKeyPress={[Function]}
>
  <label
    className="MuiFormLabel-root-13 MuiFormLabel-error-16 MuiFormLabel-filled-17 MuiInputLabel-root-6 MuiInputLabel-formControl-7 MuiInputLabel-animated-10 MuiInputLabel-shrink-9"
    data-shrink={true}
    htmlFor={undefined}
  >
    Date
  </label>
  <div
    className="MuiInputBase-root-33 MuiInput-root-20 MuiInput-underline-24 MuiInputBase-error-39 MuiInput-error-25 MuiInputBase-formControl-34 MuiInput-formControl-21"
    onClick={[Function]}
  >
    <input
      aria-invalid={true}
      autoComplete={undefined}
      autoFocus={undefined}
      className="MuiInputBase-input-43 MuiInput-input-28"
      defaultValue={undefined}
      disabled={false}
      id={undefined}
      mask={null}
      name="createdAt"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      pipe={null}
      placeholder={undefined}
      readOnly={true}
      required={false}
      rows={undefined}
      type="text"
      value="Unknown"
    />
  </div>
  <p
    className="MuiFormHelperText-root-50 MuiFormHelperText-error-51 MuiFormHelperText-filled-56"
    id={undefined}
  >
    Invalid Date Format
  </p>
</div>
`;
